src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1170,20 +1174,26 @@ private void ApplyTextDocumentChange(DocumentId documentId, SourceText newText)
 
         protected override void ApplyDocumentInfoChanged(DocumentId documentId, DocumentInfo updatedInfo)
         {
-            var document = CurrentSolution.GetDocument(documentId);
+            var document = CurrentSolution.GetRequiredDocument(documentId);
 
             FailIfDocumentInfoChangesNotSupported(document, updatedInfo);
 
             if (document.Name != updatedInfo.Name)
             {
                 GetProjectData(updatedInfo.Id.ProjectId, out var _, out var dteProject);
 
+                if (document.FilePath == null)
```

Comment:
>if (document.FilePath == null) [](start = 16, length = 30)

Contract.ThrowIfNull()?

Reply:
I'm preserving behavior here. Sorry meant to comment on this one before requesting review:

Previously FindItemByPath could have taken the null, which would then have resulted in projectItemForDocument being null and the Debug.Fail + return pattern being applied there. Rather than annotate FindItemByPath as null-accepting (which would have only been to service this code here -- the rest of the callers were fine) I elected to maintain the pattern.

My preference would be to actually fix this code, which I suspect we'll have to do anyways soon enough since this would be pretty bad dogfooding wise as we move more Roslyn stuff to shared projects.

Reply:
Put another way: I would actually prefer ThrowIfNull, but I don't know how much "in the wild" this is happening right now so didn't want to regress something if I could avoid it.

Reply:
Yes, that's a risk. If this silently fails isn't it masking a bug though? Would it be worth calling `FatalError.ReportWithoutCrash`?

Reply:
So the general brokeness of this code is already tracked by that other bug; I imagine the function will change entirely once that happens. ReportWithoutCrash seems wise though.